lib/pure/terminal.nim

```diff
@@ -677,3 +731,14 @@ when not defined(testing) and isMainModule:
   stdout.setForeGroundColor(fgBlue)
   stdout.writeLine("ordinary text")
   stdout.resetAttributes()
+
+when defined(windows):
+  var mode: DWORD = 0
+  discard getConsoleMode(getStdHandle(STD_OUTPUT_HANDLE), addr(mode))
+  mode = mode or ENABLE_VIRTUAL_TERMINAL_PROCESSING
+  trueColorIsSupported = setConsoleMode(getStdHandle(STD_OUTPUT_HANDLE), mode) != 0
+else:
+  when compileOption("taintmode"):
+    trueColorIsSupported = string(getEnv("COLORTERM")).toLowerAscii() in ["truecolor", "24bit"]
```

Comment:
This version works for a disabled taint mode too.

Reply:
Done.